letsencrypt/client/display/ops.py

```diff
@@ -0,0 +1,140 @@
+"""Contains UI methods for LE user operations."""
+import os
+import sys
+
+import zope.component
+
+from letsencrypt.client import interfaces
+from letsencrypt.client.display import util as display_util
+
+# Define a helper function to avoid verbose code
+util = zope.component.getUtility  # pylint: disable=invalid-name
+
+
+def choose_authenticator(auths, errs):
+    """Allow the user to choose their authenticator.
+
+    :param list auths: Where each is a tuple of the form
+        ('description', 'IAuthenticator') where IAuthenticator is a
```

Comment:
setting up class variable `description` for authenticators, as advised in https://github.com/letsencrypt/lets-encrypt-preview/pull/250/files#r25097223, would simplify the code a lot and get rid of those cryptic tuples.